src/Solve.cpp

```diff
@@ -44,18 +44,22 @@ class SolveExpression : public IRMutator {
         map<Expr, CacheEntry, ExprCompare>::iterator iter = cache.find(e);
         if (iter == cache.end()) {
             // Not in the cache, call the base class version.
-            debug(4) << "Mutating " << e << " (" << uses_var << ")\n";
+            debug(4) << "Mutating " << e << " (" << uses_var << ", " << failed << ")\n";
             bool old_uses_var = uses_var;
```

Comment:
I sure which C++ had a terse way to express this pattern (that wasn't an RIAA class)...